libc/string/string_tokpbrkstr.c

```diff
@@ -0,0 +1,695 @@
+	char *tok;
+	TEST_ASSERT_EQUAL_PTR(str, (tok = strtok(str, separators)));
+	TEST_ASSERT_EQUAL_STRING("Lor", tok);
+	TEST_ASSERT_EQUAL_PTR(NULL, (tok = strtok(NULL, separators)));
+}
+
+
+TEST(string_tok, empty_args)
+{
+	char empty[] = { 0 };
+
+	/* Different scenarios of acquiring null pointer as output*/
+	TEST_ASSERT_EQUAL_PTR(NULL, strtok("", "d"));
+	TEST_ASSERT_EQUAL_PTR(NULL, strtok(NULL, ""));
+
+	TEST_ASSERT_EQUAL_PTR(NULL, strtok(empty, ""));
+	TEST_ASSERT_EQUAL_PTR(NULL, strtok(NULL, ""));
+
+	TEST_ASSERT_EQUAL_PTR(NULL, strtok(empty, "\0"));
+	TEST_ASSERT_EQUAL_PTR(NULL, strtok(NULL, "\0"));
+
+	TEST_ASSERT_EQUAL_PTR(NULL, strtok("abc\0", "abc"));
+	TEST_ASSERT_EQUAL_PTR(NULL, strtok(NULL, "abc"));
+}
+
+
+TEST(string_tok, multi_call)
+{
+	char loremStr[20] = LOREM_IPSUM,
+		 asciiSep[2] = { 0 },
+		 sep[] = "ImDr";
+	int i;
+
+	/*
+	 * In this case we are checking if it is possible to
+	 * pass different stop points in each call to
+	 * get desired tokens
+	 */
+	TEST_ASSERT_EQUAL_PTR(loremStr, strtok(loremStr, "I"));
+	TEST_ASSERT_EQUAL_PTR(&loremStr[7], strtok(NULL, "m"));
+	TEST_ASSERT_EQUAL_PTR(&loremStr[11], strtok(NULL, "D"));
+	TEST_ASSERT_EQUAL_PTR(&loremStr[13], strtok(NULL, "r"));
+	TEST_ASSERT_EQUAL_PTR(NULL, strtok(NULL, sep));
+
+	/*
+	 * strtok() changes the content of a string,
+	 * that's why we have to set it once again
+	 */
+	memcpy(loremStr, LOREM_IPSUM, sizeof(LOREM_IPSUM));
+
+	/*
+	 * After passing all delimiters we cannot avoid misplaced tokens because
+	 * strtok looks for all elements in "ImDr" and breaks a string
+	 * in points where the first occurrence found
+	 */
+	TEST_ASSERT_EQUAL_PTR(loremStr, strtok(loremStr, sep));
+	TEST_ASSERT_EQUAL_PTR(&loremStr[3], strtok(NULL, sep));
+	TEST_ASSERT_EQUAL_PTR(&loremStr[5], strtok(NULL, sep));
+	TEST_ASSERT_EQUAL_PTR(&loremStr[7], strtok(NULL, sep));
+	TEST_ASSERT_EQUAL_PTR(&loremStr[11], strtok(NULL, sep));
+	TEST_ASSERT_EQUAL_PTR(&loremStr[13], strtok(NULL, sep));
+	TEST_ASSERT_EQUAL_PTR(NULL, strtok(NULL, sep));
+
+	char *asciiStr;
+	asciiStr = testdata_createCharStr(ASCII_LENGTH);
+
+	/*
+	 * In this case, we tokenize the whole ASCII set using jump-over by
+	 * 2 elements. We are forced to jump over 2 places because strtok set
+	 *  divider placed as a null term and in this case, we are unable to
+	 * cover the whole set. Jumping over 2 places guarantees us output in the form
+	 * of one element form ascii set to do an assert with the place where
+	 * strtok has saved the stopping place
+	 */
+	TEST_ASSERT_EQUAL_PTR(&asciiStr[1], strtok(&asciiStr[1], "\2"));
+	for (i = 4; i < ASCII_LENGTH; i += 2) {
+		asciiSep[0] = i;
+		TEST_ASSERT_EQUAL_PTR(&asciiStr[i - 1], strtok(NULL, &asciiSep[0]));
+	}
+
+	/* Checking if set is empty */
+	TEST_ASSERT_EQUAL_PTR(NULL, strtok(NULL, &asciiSep[0]));
+	free(asciiStr);
+}
+
+
+TEST(string_tok, out_of_ascii)
+{
+	char *extAsciiStr = create_extAscii_set();
+
+	char divider[6] = { 0 },
+		 separator[ASCII_LENGTH];
```

Comment:
not sure if it's worth it, but if you want:
```suggestion
	char divider[6] = { 0 },
		 separator[ASCII_LENGTH];
		 *extAsciiStr = create_extAscii_set();
```